agent/api/eniattachment.go

```diff
@@ -90,6 +90,15 @@ func (eni *ENIAttachment) StopAckTimer() {
 	eni.ackTimer.Stop()
 }
 
+// HasExpired returns true if the ENI attachment object has exceeded the
+// threshold for notifying the backend of the attachment
+func (eni *ENIAttachment) HasExpired() bool {
+	eni.guard.RLock()
```

Comment:
When does 'expiresAt' change? Do you need this?

Reply:
Whenever ACS sends a new message. It's better to be safe than sorry.

Reply:
That makes sense! I was didn't realize expiresAt was something that would be constantly updated

Reply:
We don't expect it. But, it may change because of bugs in ACS for example.